### PR TITLE
[enh] Refactoring of package download process

### PR DESCRIPTION
**Pull request by zamentur**
## The problem

Specific process is used for github, but it's not anymore relevant. 
Repo listed in apps.json from an other forge download all the commit of the branch instead of just the revision needed.

## Solution

Refactor a bit and use git init and git fetch to download only what we want.

## PR Status

Ready

## How to test
Try to do a yunohost app install with appname or app repo

### CODE REVIEW

**Review by kay0u — APPROVED** [review]
You did it!! :heart:

**Comment by alexAubin** [comment]
Yolomerging

File: src/yunohost/app.py [hunk 1]
```diff
@@ -2237,61 +2238,15 @@ def _fetch_app_from_git(app):
...
-        try:
-            revision = _get_git_last_commit_hash(url, branch)
-        except Exception as e:
-            logger.debug("cannot get last commit hash because: %s ", e)
-        else:
-            manifest['remote']['revision'] = revision
+        tree_index = url.rfind('/tree/')
+        if tree_index > 0:
+            url = url[:tree_index]
+            branch = app[tree_index + 6:]
```
Comment (Psycojoker) [inline src/yunohost/app.py:2248]: I'm a bit late but this snippet could have been written:

```python
if "/tree/" in url:
    url, branch = url.split("/tree/", 1)
```

Comment (alexAubin) [reply src/yunohost/app.py:2248]: Integrated in https://github.com/YunoHost/yunohost/commit/d5f8eb06c8996d66ae8de79a863241a1192b2742 :+1:

Comment (Psycojoker) [reply src/yunohost/app.py:2248]: Thx ❤️

**Pull request merged**